src/network/peers.rs

```diff
@@ -114,14 +114,16 @@ pub struct Peers<N: Network, E: Environment> {
     seen_outbound_connections: RwLock<HashMap<SocketAddr, SystemTime>>,
     /// The map of peers to a map of transaction IDs to their last seen timestamp.
     seen_outbound_transactions: RwLock<HashMap<SocketAddr, HashMap<N::TransactionID, SystemTime>>>,
+    /// The list of tasks spawned by the node.
+    tasks: Tasks<task::JoinHandle<()>>,
```

Comment:
Can someone clarify why a new tasks struct is being introduced here? One on the `server.rs` level is already available and passed through. For `update` functionality, the existing `Tasks` can already be passed through as needed there as well

Reply:
it's not strictly necessary - it only reduces some shallow cloning, but that shouldn't be too much of an issue; I added a commit removing it from `Peers`